src/commands/project/append-variable-group.ts

```diff
@@ -0,0 +1,175 @@
+import commander from "commander";
+import {
+  build as buildCmd,
+  exit as exitCmd,
+  populateInheritValueFromConfig,
+  validateForRequiredValues,
+} from "../../lib/commandBuilder";
+import {
+  hasValue,
+  validateProjectNameThrowable,
+  validateOrgNameThrowable,
+} from "../../lib/validator";
+import { Bedrock, Config } from "../../config";
+import { logger } from "../../logger";
+import { build as buildError, log as logError } from "../../lib/errorBuilder";
+import { errorStatusCode } from "../../lib/errorStatusCode";
+import decorator from "./append-variable-group.decorator.json";
+import { BedrockFileInfo, BedrockFile } from "../../types";
+import * as bedrockYaml from "../../lib/bedrockYaml";
+import { hasVariableGroup } from "../../lib/pipelines/variableGroup";
+import { AzureDevOpsOpts } from "../../lib/git";
+import { appendVariableGroupToPipelineYaml } from "../../lib/fileutils";
+import { SERVICE_PIPELINE_FILENAME } from "../../lib/constants";
+
+// Values that need to be pulled out from the command operator
+export interface CommandOptions {
+  orgName: string | undefined;
+  personalAccessToken: string | undefined;
+  devopsProject: string | undefined;
+}
+
+// Configuration values
+interface ConfigValues {
+  orgName: string;
+  personalAccessToken: string;
+  devopsProject: string;
+}
+
+/**
+ * Validates values passed as options
+ * @param opts The initialized options
+ */
+export const validateValues = (opts: CommandOptions): ConfigValues => {
+  populateInheritValueFromConfig(decorator, Config(), opts);
+  validateForRequiredValues(decorator, opts, true);
+
+  // validateForRequiredValues already check required values
+  // || "" is just to satisfy eslint rule.
+  validateProjectNameThrowable(opts.devopsProject || "");
+  validateOrgNameThrowable(opts.orgName || "");
+
+  return {
+    orgName: opts.orgName || "",
+    personalAccessToken: opts.personalAccessToken || "",
+    devopsProject: opts.devopsProject || "",
+  };
+};
+
+/**
+ * Check project dependencies
+ * @param projectPath Path to the project directory
+ */
+export const checkDependencies = (projectPath: string): void => {
+  const fileInfo: BedrockFileInfo = bedrockYaml.fileInfo(projectPath);
+  if (fileInfo.exist === false) {
+    throw buildError(
+      errorStatusCode.VALIDATION_ERR,
+      "project-append-variable-group-cmd-err-dependency"
```

Comment:
i suggest `project-append-variable-group-cmd-err-dependency` to be `project-append-variable-group-cmd-err-bedrock-yaml-missing`